crates/sim2h/src/sim2h_im_state.rs

```diff
@@ -882,9 +882,14 @@ impl StoreHandle {
         .boxed()
     }
 
-    pub fn spawn_new_connection(&self, space_hash: SpaceHash, agent_id: AgentId, uri: Lib3hUri) {
+    pub async fn spawn_new_connection(
+        &self,
+        space_hash: SpaceHash,
+        agent_id: AgentId,
+        uri: Lib3hUri,
+    ) {
         let f = self.new_connection(space_hash, agent_id, uri);
-        tokio::task::spawn(f);
+        f.await
```

Comment:
My intention was that functions prefixed with `spawn_` would not return futures (or anything else) because they spawn the task.

If we don't need this, we can just delete this whole function - and lib.rs can call `new_connection()` directly : )